app/src/main/java/org/sagebionetworks/research/crf/CrfSample.java

```diff
@@ -74,12 +72,31 @@ private void addTask(final ViewGroup taskContainer, final Intent taskIntent,
         taskButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                startTask(taskIntent);
+                startTask(taskTitle);
             }
         });
     }
 
-    private void startTask(final Intent taskIntent) {
+    private void startTask(final String taskTitle) {
+
+        Intent taskIntent = null;
+        switch (taskTitle) {
+            case trainingTaskTitle:
+                taskIntent =  CrfTaskIntentFactory.getHeartRateTrainingTaskIntent(this);
+                break;
+            case restingHrTaskTitle:
+                taskIntent = CrfTaskIntentFactory.getHeartRateMeasurementTaskIntent(this);
+                break;
+            case snapshotTaskTitle:
+                if (gender != null && birthYear != null) {
+                    taskIntent = CrfTaskIntentFactory
```

Comment:
@nategbrown9 I added an additional function to the intent factory that lets you pass in the previously answered questions to gender and birthYear and those questions get skipped the next time you do them. This was updated to match how the flow already was on iOS.